activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -118,7 +120,11 @@ def commit
         connection.release_savepoint(savepoint_name)
         super
         parent = connection.transaction_manager.current_transaction
-        records.each { |r| parent.add_record(r) }
+        if parent.uncommittable?
+          commit_records
```

Comment:
We want to communicate that the app has a logical commit boundary for AR instances that may differ from the physical db commit boundary. "parent.uncommittable?" reads as "parent db txn cannot be committed" when we want to know "should this txn commit its AR instances?" Could rename to something like `participates_in_records_transaction` which defaults to `true`.

Also: should this class even be responsible for figuring out whether it should commit records? Reaching in to the txn manager to check state is a red flag, not to mention that it relies on txn manager having already popped this txn off its stack leaving the parent as the current txn. Perhaps this should be the TransactionManager's duty in the first place:
```ruby
def commit_transaction
  transaction = @stack.pop
  transaction.commit
  transaction.commit_records if !current_transaction.participates_in_records_transaction?
end
```